src/physics/em/detail/MollerBhabhaInteractor.i.hh

```diff
@@ -0,0 +1,219 @@
+{
+    // Allocate memory for the produced electron
+    Secondary* electron_secondary = this->allocate_(1);
+
+    if (electron_secondary == nullptr)
+    {
+        // Fail to allocate space for a secondary
+        return Interaction::from_failure();
+    }
+
+    // Temporary
+    (void)sizeof(rng);
+
+    // Maximum transferable energy to the free electron
+    real_type max_kinetic_energy;
+
+    if (inc_particle_is_electron_)
+    {
+        // Moller scattering (e-e-)
+        max_kinetic_energy = 0.5 * inc_energy_.value();
+    }
+    else
+    {
+        // Bhabha scattering (e+e-)
+        max_kinetic_energy = inc_energy_.value();
+    }
+
+    // Set up sampling parameters
+    real_type total_energy = inc_energy_.value() + shared_.electron_mass_c_sq;
+    real_type x_min = shared_.min_valid_energy_.value() / inc_energy_.value();
+    real_type x_max = max_kinetic_energy / inc_energy_.value();
+    real_type gamma = total_energy / shared_.electron_mass_c_sq;
+    real_type gamma_sq = gamma * gamma;
+    real_type beta_sq  = 1.0 - (1.0 / gamma_sq);
+    real_type x;
+    real_type z;
+    real_type rejection_function_g;
+    real_type random[2];
+
+    //// Moller scattering (e-e-) ////
+    if (inc_particle_is_electron_)
+    {
+        real_type two_gamma_term = (2.0 * gamma - 1.0) / gamma_sq;
+        real_type y              = 1.0 - x_max;
+        rejection_function_g     = 1.0 - two_gamma_term * x_max
+                               + x_max * x_max
+                                     * (1.0 - two_gamma_term
+                                        + (1.0 - two_gamma_term * y) / (y * y));
+
+        do
+        {
+            random[0] = generate_canonical(rng);
+            random[1] = generate_canonical(rng);
+
+            x = x_min * x_max / (x_min * (1.0 - random[0]) + x_max * random[0]);
+            y = 1.0 - x;
+            z = 1.0 - two_gamma_term * x
+                + x * x
+                      * (1.0 - two_gamma_term
+                         + (1.0 - two_gamma_term * y) / (y * y));
+        } while (rejection_function_g * random[1] > z);
+    }
+
+    //// Bhabha scattering (e+e-) ////
+    else
+    {
+        real_type y    = 1.0 / (1.0 + gamma);
+        real_type y2   = y * y;
+        real_type y12  = 1.0 - 2.0 * y;
+        real_type b1   = 2.0 - y2;
+        real_type b2   = y12 * (3.0 + y2);
+        real_type y122 = y12 * y12;
+        real_type b4   = y122 * y12;
+        real_type b3   = b4 + y122;
+
+        y = x_max * x_max;
+
+        rejection_function_g
+            = 1.0
+              + (y * y * b4 - x_min * x_min * x_min * b3 + y * b2 - x_min * b1)
+                    * beta_sq;
+        do
+        {
+            random[0] = generate_canonical(rng);
+            random[1] = generate_canonical(rng);
+
+            x = x_min * x_max / (x_min * (1.0 - random[0]) + x_max * random[0]);
+            y = x * x;
+            z = 1.0 + (y * y * b4 - x * y * b3 + y * b2 - x * b1) * beta_sq;
```

Comment:
All of your `b` variables earlier were functions of `y`. Do they need to be updated?